### ISSUE TITLE
Problems with Juanma's Model

### ISSUE THREAD

**Issue by jmbilbao** [CityScope/CS_Dynamic_Urban_Planning#3]
Mireia,

I have some trouble in **line 373**. The program stops after about 4000 cycles, and collapses in that line. I have been trying to find the problem, but I am not able to solve it.

On the other hand, I would be needing some help with the way of introducing new agents that come from outside of Kendall

Thank you!!!

**Comment by mireiayurrita** [comment]
Hi Juanma. Whar do you mean by "collapses in that line"?? This is what I just got:
![image](https://user-images.githubusercontent.com/58276088/108103539-622f5900-708a-11eb-9d36-c7c4ccce6857.png)

As for the introduction of agents that come from outside Kendall, that is what the cityscopable version  was doing with entry points. Basically you can create a boolean attribute to ``people agents'' that define them as outsiders or not. People who live in kendall can be assigned a building in iteration 0 while the outsiders are assigned an entry point. That is what I did, I guess there are many more possibilities but that worked for me. 
 (I assign the issue to myself and Arnaud so that he keeps track of what we are saying. Btw, whenever you change things in the model do not hesitate to push it so that we are all on the same page).

**Comment by agrignard** [comment]
Hi @jmbilbao did you push your version, I can see if I have the same issue also

**Comment by jmbilbao** [comment]
Thank you Mireia! I'll try what you said for agents from outside of Kendall

Sorry, I meant that the execution of the program stops at approximately 40000 cycles (if you multiply the step by 10, it will step at 4000) at the mentioned line 373.

Arnaud, the version is the one that is called Juanma Model. Thank you!

**Comment by mireiayurrita** [comment]
Hi Juanma, first I tried to run the cityscopable and the gameit models to make sure this error was not inherited. I attach the screenshots so that you can follow my reasoning when trying to identify the bug. 
Cityscopable. 80000 cycles. No error.
![image](https://user-images.githubusercontent.com/58276088/108225245-0c1aee00-713c-11eb-92a0-5393ca084076.png)

CityScopable. More than one day (prove it is not the daily schedule problem). No error.
![image](https://user-images.githubusercontent.com/58276088/108225276-1210cf00-713c-11eb-8c79-ee6ee38a388c.png)

GameIt. More than 4000 cycles (step much bigger). Many days. No problem with daily schedule. No problem.
![image](https://user-images.githubusercontent.com/58276088/108225332-2359db80-713c-11eb-9088-3183de7027f6.png)


Juanma. Step 1 sec. 10000 cycles it gets stuck. ?????? No error message pops up.
![image](https://user-images.githubusercontent.com/58276088/108225457-3f5d7d00-713c-11eb-8d36-d6cd0dfea0b0.png)


Juanma. Step 10 secs. Gets stuck and error  message pops up. Check which is the people agent and bus_stop  concerned. Very useful to inspect each agent and see their attributes real-time. 
Error given by bus 1. Stop_passengers[bus_stop(9)] is nil. The thing is that bus 1 covers route 69 and bus_stop(9) is from 68, so it is not in the map stop_passengers of bus 1. In gameIt there was only a bus so this was not happening.
![image](https://user-images.githubusercontent.com/58276088/108225486-46848b00-713c-11eb-87c8-41f34bb42bbc.png)

In cityscopable this is avoided making sure that the routes coincide. Notice the difference when defining b <- bus_stop …….
 
![image](https://user-images.githubusercontent.com/58276088/108225512-4c7a6c00-713c-11eb-99c1-aa6861231d5d.png)


However, @agrignard , why with step 1#sec it gets stuck but no error message pops up???
@jmbilbao , make sure that the merging process you made btw cityscopable and game_it is correctly set up. Buses and T-s need to be set up as in cityscopable, since this kind of errors were already tackled there.

**Comment by jmbilbao** [comment]
Yes, i had already looked at the problems with the buses. However, I have the same problem, even if i set the step in 1 second
![image](https://user-images.githubusercontent.com/72378554/108251566-d7b52b00-7157-11eb-8cdd-3bdbc8c9451b.png)

**Comment by mireiayurrita** [comment]
Okay. Perfect. Then one mystery less. I push the corrected version (I am assuming you did not change anything with respect to what I pushed on Feb6). 
This is what I got after a 24h in the simulation (step 10 secs). It seems there is no error. However, as I told the other day, if your idea is to continue with the 1km x 1 km scope of Kendall, it might be interesting to get rid of the bus stops that are outside of that area  with QGIS so that the buses don't get crazy on what topology to follow. Animo!
![image](https://user-images.githubusercontent.com/58276088/108259016-f0760e80-7160-11eb-8f46-e8409f7c64e9.png)
